### PR TITLE
feat: add Tag component with customizable styles for issue #696

### LINKED ISSUE #696: Tag Component
refer https://github.com/starknet-id/starknet.quest/issues/693

### PR DESCRIPTION
**Pull request by coxmars**
## Pull Request type

- Feature: This PR adds a new Tag component to the UI library for displaying a tag with customizable styles.

- Resolves: #696 

## Questions
- Are the props I used correct or should I modify/add others?

## Other information

- Added Tag component in components/UI/tag.tsx
- Supports custom colors and styles

image

## Summary by CodeRabbit

- **New Features**
  - Introduced `QuestTag` component for rendering customizable tags with labels and icons.
  - Updated `FeaturedQuest` and `Quest` components to use `QuestTag` for displaying quest details and completion status.
  - Added `backgroundColor` prop to `CheckIcon` and `UnavailableIcon` components.

- **Refactor**
  - Refactored reward rendering logic in `Quest` and `reward` components to use `QuestTag`.

- **Style**
  - Minor style adjustments and cleanup in various components.

- **Chores**
  - Removed unused imports and updated export statements for better code clarity.

### CODE REVIEW

**Review by ayushtom — CHANGES_REQUESTED** [review]
Hey 
can you also replace this in the current app wherever we implement a tag?
on merging then your compoenent will be used in production

**Comment by coxmars** [comment]
> Hey can you also replace this in the current app wherever we implement a tag? on merging then your compoenent will be used in production

All right, I'll let you know as soon as I do.

**Comment by coxmars** [comment]
Hi @ayushtom I already replaced the old tag for the new tag component in the code.

- Comparison -  left tag (old) and right tag new one

Comparison


- Cards

Tags

- Responsive

ResponsiveTags

File: app/quest/[questPage]/quest.tsx [hunk 1]
```diff
@@ -130,18 +131,7 @@ const Quest: FunctionComponent<QuestPageProps> = ({
           <BackButton onClick={() => router.back()} />
         </div>
         <div className={styles.imageContainer}>
-          {quest.issuer === "loading" ? (
```
Comment (ayushtom) [inline app/quest/[questPage]/quest.tsx:133]: why remove the skeleton?

File: components/UI/featured_banner/featuredQuest.tsx [hunk 2]
```diff
@@ -62,19 +63,10 @@ const FeaturedQuest: FunctionComponent<FeaturedQuestProps> = ({
...
-              <CDNImage
-                alt={"Feature Quest Reward"}
-                height={20}
-                width={20}
-                src={issuer?.logoFavicon}
-                className={styles.featuredQuestRewardIcon}
-              />
-              <p className={styles.featuredQuestReward}>{reward}</p>
-            </div>
-          ) : null}
```
Comment (ayushtom) [inline components/UI/featured_banner/featuredQuest.tsx:78]: please replace the div tag inside the braces here. I think removing null states is not required

File: components/UI/iconsComponents/icons/checkIcon.tsx [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
-const CheckIcon: FunctionComponent<IconProps> = ({ width, color }) => {
+const CheckIcon: FunctionComponent<IconProps> = ({ width, color, backgroundColor }) => {
   return (
     <svg
       xmlns="http://www.w3.org/2000/svg"
       width={width}
       height={width}
       viewBox="0 0 24 24"
       fill="none"
+      style={{ backgroundColor: backgroundColor }}
```
Comment (ayushtom) [inline components/UI/iconsComponents/icons/checkIcon.tsx]: can be written as `style={{backgroundColor}}`

File: components/UI/iconsComponents/icons/unavailableIcon.tsx [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 
-const UnavailableIcon: FunctionComponent<IconProps> = ({ width, color }) => {
+const UnavailableIcon: FunctionComponent<IconProps> = ({ width, color, backgroundColor }) => {
   return (
     <svg
       width={width}
       height={width}
       viewBox="0 0 24 24"
       xmlns="http://www.w3.org/2000/svg"
+      style={{ backgroundColor: backgroundColor }}
```
Comment (ayushtom) [inline components/UI/iconsComponents/icons/unavailableIcon.tsx]: same as above

File: app/quest/[questPage]/quest.tsx (hunk 1 above)
Comment (coxmars) [reply app/quest/[questPage]/quest.tsx:133]: It was an error, I fixed it

**Comment by coxmars** [comment]
Hello @ayushtom,

I've made the changes you requested. If there's anything else that needs to be adjusted, please let me know. Thank you for your feedback!

**Comment by ayushtom** [comment]
Original Tag
Screenshot 2024-05-31 at 2 23 11 PM

Current Tag
Screenshot 2024-05-31 at 2 23 22 PM


Hey i think visually there are some differences in the component. Can you fix this and commit the changes please

**Comment by coxmars** [comment]
> Original Tag Screenshot 2024-05-31 at 2 23 11 PM
> 
> Current Tag Screenshot 2024-05-31 at 2 23 22 PM
> 
> Hey i think visually there are some differences in the component. Can you fix this and commit the changes please

Hi @ayushtom, I already applied this changes look at this:

Current Tag

DoneTag

- Improved the icon border color and the position when is Done/Expired, now all use cases are covered

**Comment by ayushtom** [comment]
i think the font looks different from what is currrently live. Please add correct font stylings

**Comment by coxmars** [comment]
> i think the font looks different from what is currrently live. Please add correct font stylings

I applied the correct font "Sora" as you can see in the below image, the first image is the old one and the second one is the improved one.

image (1)

**Comment by coxmars** [comment]
Hi @ayushtom, I already resolved merge conflicts

**Review by ayushtom — APPROVED** [review]
lgtm

**Pull request merged**